src/klein/test/test_app.py

```diff
@@ -65,6 +65,31 @@ def anotherInstance(self):
 
 
 
+class DuplicateHasher(object):
+    """
+    Every L{DuplicateHasher} has the same hash value and compares equal to
```

Comment:
Why not use an unhashable class instead?  That's this PR's stated goal.

I understand that this tests a stronger property, but if you construct an unhashable class with `attrs`, then its `__hash__` and `__eq__` methods won't need coverage.

Reply:
They both test effectively the same property, but this tests a much worse failure mode than just raising an exception :).